### PR TITLE
[add] py-metomi-isodatetime: new recipe, required by py-cylc-flow

### PR DESCRIPTION
**Pull request by LydDeb**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage and version sha256.  Adding @adamjstewart

**Review by manuelakuhn — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyMetomiIsodatetime(PythonPackage):
+    """Python ISO 8601 date time parser and data model/manipulation utilities."""
+
+    homepage = "https://github.com/metomi/isodatetime"
+    url = "https://github.com/metomi/isodatetime/archive/refs/tags/3.0.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py (hunk 1 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py]: @LydDeb, I guess Adam wants you to change the package to pypi:

https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

**Review by manuelakuhn — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py (hunk 1 above)
Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py]: Thank you @bernhardkaindl
I first tried with pypi but the fetch didn't go well. This is why I put the URL to github. I have changed my working environment in the meantime and I no longer reproduce this fetch problem with pypi.

File: var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyMetomiIsodatetime(PythonPackage):
+    """Python ISO 8601 date time parser and data model/manipulation utilities."""
+
+    homepage = "https://github.com/metomi/isodatetime"
+    pypi = "metomi-isodatetime/metomi-isodatetime-1!3.0.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py:13]: Does this actually work, or does it replace `1!3.0.0` with `3.0.0`? We might need a `url_for_version` function. You can test this by running `spack fetch py-metomi-isodatetime` or `spack info py-metomi-isodatetime` to see what URL it uses.

Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-metomi-isodatetime/package.py:13]: This works. It uses `https://files.pythonhosted.org/packages/source/m/metomi-isodatetime/metomi-isodatetime-1!3.0.0.tar.gz`

**Review by bernhardkaindl — APPROVED** [review]
@LydDeb confirmed that the pypi download works.

**Comment by LydDeb** [comment]
@adamjstewart Do I need to make any other changes?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**